docs/advanced_usage/how_to_auxiliary.rst

```diff
@@ -157,129 +110,180 @@ See below an example implementing the basic functionalities of a Thread Auxiliar
+          self.queueing_event = threading.Event()
+          self.collect_messages = functools.partial(_collect_messages, com_aux=self)
+
+      @open_connector
+      def _create_auxiliary_instance(self) -> bool:
+          """Open the connector communication.
+
+          :return: True if the channel is correctly opened otherwise False
+          """
+          log.internal_info("Auxiliary instance created")
+          return True
+
+      @close_connector
+      def _delete_auxiliary_instance(self) -> bool:
+          """Close the connector communication.
+
+          :return: always True
+          """
+          log.internal_info("Auxiliary instance deleted")
+          return True
+
+      def send_message(self, raw_msg: bytes) -> bool:
+          """Send a raw message (bytes) via the communication channel.
+
+          :param raw_msg: message to send
+
+          :return: True if command was executed otherwise False
+          """
+          return self.run_command("send", raw_msg)
+
+      def run_command(
+          self,
+          cmd_message: Any,
+          cmd_data: Any = None,
+          blocking: bool = True,
+          timeout_in_s: int = None,
+      ) -> bool:
+          """Send a request by transmitting it through queue_in and
+          populate queue_tx with the command verdict (successful or not).
+
+          :param cmd_message: command to send
+          :param cmd_data: data you would like to populate the command
+              with
+          :param blocking: If you want the command request to be
+              blocking or not
+          :param timeout_in_s: Number of time (in s) you want to wait
+              for an answer
+
+          :return: True if the request is correctly executed otherwise
+              False
+          """
+          with self.lock:
+              log.internal_debug(
+                  f"sending command '{cmd_message}' with payload {cmd_data} using {self.name} aux."
+              )
+              state = None
+              self.queue_in.put((cmd_message, cmd_data))
+              try:
+                  state = self.queue_tx.get(blocking, timeout_in_s)
+                  log.internal_debug(
+                      f"command '{cmd_message}' successfully sent for {self.name} aux"
+                  )
+              except queue.Empty:
+                  log.error(
+                      f"no feedback received regarding request {cmd_message} for {self.name} aux."
+                  )
+          return state
+
+      def receive_message(
+          self,
+          blocking: bool = True,
+          timeout_in_s: float = None,
+      ) -> Optional[bytes]:
+          """Receive a raw message.
+
+          :param blocking: wait for message till timeout elapses?
+          :param timeout_in_s: maximum time in second to wait for a response
+
+          :returns: raw message
+          """
+
+          # Evaluate if we are in the context manager or not
+          in_ctx_manager = False
```

Comment:
This is way too specific for a tutorial on how to implement an auxiliary, we should take the CommunicationAuxiliary code but rather keep the previous example and simply adapt it